lib/Command/RecreateMasterKey.php

```diff
@@ -187,41 +125,35 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 						$this->rootView->deleteAll($filesEncryptionDir . '/files_encryption');
 					}
 
-					$this->appConfig->setValue('core', 'encryption_enabled', 'no');
-					$this->appConfig->deleteKey('encryption', 'useMasterKey');
-					$this->appConfig->deleteKey('encryption', 'masterKeyId');
-					$this->appConfig->deleteKey('encryption', 'recoveryKeyId');
-					$this->appConfig->deleteKey('encryption', 'publicShareKeyId');
-					$this->appConfig->deleteKey('files_encryption', 'installed_version');
+					$this->util->removeEncryptionAppSettings();
 				}
 				$output->writeln("\nDecryption completed\n");
 
 				//Reencrypt again
 				$this->appManager->enableApp('encryption');
 				$this->appConfig->setValue('core', 'encryption_enabled', 'yes');
-				$this->appConfig->setValue('encryption', 'enabled', 'yes');
+				$this->appConfig->setValue('encryption', 'useMasterKey', '1');
+
 				$output->writeln("Encryption started\n");
```

Comment:
This message might be misleading. The encryption hasn't started at that point.

Reply:
Moved to line above `encryptAll` is called.